### DISCUSSION TITLE
Converting Legacy Code Helpdesk Plugin Circa 2007 to work with Latest E107 and PHP 8.2

### DISCUSSION THREAD

**Discussion by BaxterSpace** [e107inc/e107#5358] [General]
RESOLVED

**Comment by Jimmi08** [comment]
Hi,
not working for PHP 8 is mainly a constant issue.  You need to fix all warnings in PHP 7 first. (PHP 8 has a blank page, fatal errors)

Database issues are related to changes in e107 db functions, they are not compatible (regardless of what they say) 

I updated the gold system (not used anymore), if you go through the commits, you will see what to change.  
https://github.com/Jimako-e107-plugins/e107-Gold-System

Do you mean this plugin? Or something else?
https://github.com/Jimako-e107-plugins/e107-old-plugins/tree/master/helpdesk3_menu_V3.4/helpdesk3_menu

If you want help from me, please create a new repo under your account and upload the original code there.  I prefer to tell you what is wrong a show you how to fix it, not do your work.  My time is limited. It is a pity that you haven't asked sooner when I updated those old plugins a lot.

**Comment by Alex-e107nl** [comment]
Hi Baxter which e107 version do you use? Sorry my bad... :-) (edited)

**Comment by BaxterSpace** [comment]


**Reply by Jimmi08** [reply]
2.3.3 is already outdated, with some bugs.  With PHP 8.2 you need to use developer version from here.

**Comment by Vodhin** [comment]
Besides issues with undefined CONSTANTS causing errors, the count() and foreach() and other array() functions now throw errors if the $variable is not an array. It boggles my mind why PHP decided to throw an error instead of returning nothing. (thinking about this, I wonder if there are php.ini settings that could be changed to change this)

**Comment by tgtje** [comment]
just _mentioning_, to be taken in consideration??. A difference also (unsure if its a 'bummer'; due to dbase problems) MyISAM vs InnoDb tables. Although it should work, but no idea iff relation to old coding methods.

**Comment by Jimmi08** [comment]
@BaxterSpace you can't use GPT for this if you are not familiar with old and code, debugging their answers related to e107...    But as soon as you help it a bit with correct way, it is big help.

Ok, I will try to do it...  I have some hours. My way with repo and issues, at first renaming and removing _menu from plugin folder. 
@BaxterSpace you will continue yourself as soon as you understand what to change.  For fully update I need testing data. You can find my email at the bottom of my site e107sk.com

**Reply by Jimmi08** [reply]
The first is the fix for PHP 7.4. and latest e107 (2.4)  

Here is the repo
https://github.com/Jimako-e107-plugins/helpdesk3

Here is the result for me after installing

![image](https://github.com/user-attachments/assets/c911b753-f6b1-4a9d-8ce0-2419999ecf1e)

![image](https://github.com/user-attachments/assets/10205df3-241d-4fe7-8304-16bb29c4153a)

I need to know what wasn't working in **PHP 7**

**Reply by Jimmi08** [reply]
Have you ever checked your database with Tools/Database/Check tables or so?  
Have you been able to send me your preferences to test them as I asked? 
Have you ever set debug mode on and checked SQL errors?   
Have you ever checked your preferences with Tools/Database/Check Preferences?  Plugins are on the second tab. 
You can have installed other plugins that messed up your preferences so they cannot be saved.

Without access to your site, admin, database, or FTP...  nobody can tell—or at least an exported database.

**Comment by BaxterSpace** [comment]


**Reply by Jimmi08** [reply]
As you see in my screenshots, it works for me - tickets are saved.  Editable. Preferences fixed. 
Your issue is either outdated core e107 (2.3.3) or your preferences I haven't tested yet or conflict with other plugins or anything.  Or not compatible theme.   I can fix the issue only if I can simulate it.  

The Helpdesk3 plugin has now separated the repo. If you want to fix it, you need to create a normal issue there with screenshots or detailed descriptions.

**Reply by Jimmi08** [reply]
only folder helpdesk3,  older version is just backup.  I let you there issues with marking source of problem -  that plugin has some bugs without e107 changes and PHP changes.  I will close them soon.

![image](https://github.com/user-attachments/assets/3a7ad595-771c-40db-9b22-82a8cb1c8d4c)

**Reply by Jimmi08** [reply]
Fresh install 2.4.0,  core bootstrap5 theme,  installed helpdesk3 , saving prefs (no change), added category, added ticket on frontend for this category. Result:     
![image](https://github.com/user-attachments/assets/d0299bfa-28e2-4c97-95cf-b19dc6809943)

**Comment by BaxterSpace** [comment]


**Reply by Jimmi08** [reply]
Are you aware that I am not a user of that plugin?  That I don't know what you have on mind? This?  It works. 

![image](https://github.com/user-attachments/assets/c6b3dbdd-7d08-4989-95b1-c990822371d5)

**Comment by LaocheXe** [comment]
Also, e107 does have a debug thing. Just have to add something to the end
of your url to show warnings/errors. That should help you debug your code.

**Reply by Jimmi08** [reply]
I did this.  So any visible error is fixed now.

**Comment by tgtje** [comment]
hmm just tried local with latest files from @Jimmi08 's repo
php 8.2.13 little older git version of 2.4  Using My SQl server (test setting), only for some quick things..
Since **unfamiliar** had to find out how to get a helpdesk and a category (in both settings used new and update to create..
after that.. no issues *not tested further. So it does function it seems. 

_Method used : uploaded into plugins, installed, configure :  (what to do and just tried) >no cat etc.. so created that and thats all_ 


As you can see my dbase server uses MySQL (installed in it) and both InnoDB and MyISAM work together in dbase..
![helpdesk](https://github.com/user-attachments/assets/adb65618-40e0-4b22-a540-7cf3f4aa04bd)

![helpdesk2](https://github.com/user-attachments/assets/1154833d-0cfd-4de2-a405-f55bffe2ebbd)

**Reply by Jimmi08** [reply]
@tgtje  THANKS!
that way New/Edit/ Delete + Confirm is how it was very often used before. There is nothing like a category list, you need to use a dropdown to see them.  Years ago I asked how to combine the old way and new admin UI - the left menu is the issue - I can easily replace helpdesk and categories files with the new admin UI, but it doesn't work with the old left menu and at first preferences has to be changed to use fully new admin UI.  
Father Barry way - to have a separate file for each admin part is very good approach, if I find a way (I had a workaround for this a long time ago) to solve the left menu issue, I could update it more.  

Engine - I just forgot to change it but e107 is doing it yourself if I am not mistaken. With some settings or only for core plugins...

**Reply by tgtje** [reply]
@Jimmi08  Your welcome, but anything you do to it, is your decision. 
I think it would be nonetheless a good plugin for sites that relate to people having issues.
It's a pity that Father Barry (Keal) (yep a real priest)  aka G4HDU stopped years ago (for those who not have read/know about him). Health gets us all.
He had a lot of usefull things. On Github some still exists ([history](https://github.com/search?q=G4HDU&type=users)). end 2019.
As for the left menu issue (unsure your idea) take a look at [these](https://github.com/e107inc/e107/tree/master/e107_languages/English/admin/help), they still work (method) , but i started with translation, but everything is all over the core..so put a hold on that.

**Comment by BaxterSpace** [comment]


**Reply by tgtje** [reply]
@BaxterSpace Well reading in on AlmaLinux, going through forum there..alma 8 or 9? 9 should support 7.4 
I can not replicate for Linux ( but using the same installment (local Wamp) running 7.4 shows no issues.

But being 7.4 to be seen as depricated, and Alma (afaic see) uses SQL version max 8.. >>> hard to say if that is a culprit ( from a external post: The issues with authentication methods were fixed in PHP 7.4 (since 7.4.4 should be working perfectly fine)
No idea if that would still be an issue with git versions..

From 7.4 to 8.3 all running, however there is influence of MySQL version (using 8.2!!!)  if it gets too low (i presume)(; only a version 5 running for old v1 installs + same by phpadmin versions). might cause conflict or misbehaviour.
Probably there's more happening than we know and unfamiliar settings...
And ca. 2008 till now, a lottttttt has changed.

**Comment by PH5HP** [comment]
Hi,

About 'Father Barry'. He is still alive and he is in my Friends List of Facebook. He is a Hamradio Operator like me, but I do not think he is a Father anymore. He give up on E107, but I can't remember what his reasons were. I know he is/was disappointed in E107 about the road they went with E107.

I will give him a shout about this plugin. What do you want to know about the plugin?

Kind regards and a Happy New Year,
Henk - ph5hp.nl
